integ-test/src/test/java/org/opensearch/sql/calcite/standalone/CalcitePPLInSubqueryIT.java

```diff
@@ -0,0 +1,353 @@
+    verifyDataRows(
+        result1,
+        rows(1003, "David", 120000),
+        rows(1002, "John", 120000),
+        rows(1000, "Jake", 100000),
+        rows(1005, "Jane", 90000),
+        rows(1006, "Tommy", 30000));
+    verifyDataRows(
+        result2,
+        rows(1003, "David", 120000),
+        rows(1002, "John", 120000),
+        rows(1000, "Jake", 100000),
+        rows(1005, "Jane", 90000),
+        rows(1006, "Tommy", 30000));
+  }
+
+  @Test
+  public void testTwoExpressionsInSubquery() {
+    JSONObject result =
+        executeQuery(
+            String.format(
+                """
+                   source = %s
+                   | where (id, name) in [
+                       source = %s | fields uid, name
+                     ]
+                   | sort  - salary
+                   | fields id, name, salary
+                   """,
+                TEST_INDEX_WORKER, TEST_INDEX_WORK_INFORMATION));
+    verifySchema(
+        result, schema("id", "integer"), schema("name", "string"), schema("salary", "integer"));
+    verifyDataRows(
+        result,
+        rows(1003, "David", 120000),
+        rows(1002, "John", 120000),
+        rows(1000, "Jake", 100000),
+        rows(1005, "Jane", 90000));
+  }
+
+  @Test
+  public void testWhereNotInSubquery() {
+    JSONObject result =
+        executeQuery(
+            String.format(
+                """
+                   source = %s
+                   | where id not in [
+                       source = %s | fields uid
+                     ]
+                   | sort  - salary
+                   | fields id, name, salary
+                   """,
+                TEST_INDEX_WORKER, TEST_INDEX_WORK_INFORMATION));
+    verifySchema(
+        result, schema("id", "integer"), schema("name", "string"), schema("salary", "integer"));
+    verifyDataRows(result, rows(1001, "Hello", 70000), rows(1004, "David", 0));
+  }
+
+  @Test
+  public void testFilterNotInSubquery() {
+    JSONObject result =
+        executeQuery(
+            String.format(
+                """
+                   source = %s id not in [
+                       source = %s | fields uid
+                     ]
+                   | sort  - salary
+                   | fields id, name, salary
+                   """,
+                TEST_INDEX_WORKER, TEST_INDEX_WORK_INFORMATION));
+    verifySchema(
+        result, schema("id", "integer"), schema("name", "string"), schema("salary", "integer"));
+    verifyDataRows(result, rows(1001, "Hello", 70000), rows(1004, "David", 0));
+  }
+
+  @Test
+  public void testTwoExpressionsNotInSubquery() {
+    JSONObject result =
+        executeQuery(
+            String.format(
+                """
+                   source = %s
+                   | where (id, name) not in [
+                       source = %s | fields uid, name
+                     ]
+                   | sort  - salary
+                   | fields id, name, salary
```

Comment:
* what if subquery include less columns `source = %s | where (id, name) not in [source = %s | fields name]`
* or more columns, `source = %s | where (id, name) not in [source = %s | fields id, name, salary]`

Reply:
check L314

Reply:
Added more tests and convert the AssertError to SemanticCheckException @penghuo